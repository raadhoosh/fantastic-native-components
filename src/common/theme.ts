import { Theme } from "../components";
import { Platform, Dimensions, PixelRatio } from "react-native";

const deviceHeight = Dimensions.get("window").height;
const deviceWidth = Dimensions.get("window").width;
const platform = Platform.OS;
const isIphoneX =
    platform === "ios" && deviceHeight === 812 && deviceWidth === 375;

const mainTheme: Theme = {
    primary: {
        light: "#4dabf5",
        main: "#2196f3",
        dark: "#1769aa",
        contrastText: "#fff",
    },
    secondary: {
        light: "#ffac33",
        main: "#ff9800",
        dark: "#b26a00",
        contrastText: "#fff",
    },
    info: {
        light: "#33c9dc",
        main: "#00bcd4",
        dark: "#ba000d",
        contrastText: "#fff",
    },
    warning: {
        light: "#ffcd38",
        main: "#ffc107",
        dark: "#b28704",
        contrastText: "#fff",
    },
    danger: {
        light: "#f6685e",
        main: "#f44336",
        dark: "#aa2e25",
        contrastText: "#fff",
    },
    success: {
        light: "#a2cf6e",
        main: "#8bc34a",
        dark: "#618833",
        contrastText: "#fff",
    },
    purple: {
        light: "#af52bf",
        main: "#9c27b0",
        dark: "#6d1b7b",
        contrastText: "#fff",
    },
    deviceWidth: deviceHeight,
    deviceHeight: deviceWidth,
    isIphoneX: isIphoneX,
    platform: platform === "android" ? "android" : "ios",
};
export default mainTheme;