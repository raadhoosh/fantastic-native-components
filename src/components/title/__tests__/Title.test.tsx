import Title from "..";
import { ThemeProvider } from "../..";
import theme from "../../../common/theme";
import { shallow, ShallowWrapper, render } from "enzyme";
import React from "react";

describe("Title", () => {
    describe("rendering", () => {
        let wrapper: ShallowWrapper;
        beforeEach(() => {
            wrapper = shallow(<ThemeProvider theme={theme}><Title>Title</Title></ThemeProvider>);
        });
        it("renders", () => {
            expect(wrapper.exists()).toBe(true);
        });
        it("toMatchSnapshot", () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});