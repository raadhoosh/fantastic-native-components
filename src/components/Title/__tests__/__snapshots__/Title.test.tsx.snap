// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title rendering toMatchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    theme={
      Object {
        "danger": Object {
          "contrastText": "#fff",
          "dark": "#aa2e25",
          "light": "#f6685e",
          "main": "#f44336",
        },
        "deviceHeight": 750,
        "deviceWidth": 1334,
        "grey": Object {
          "contrastText": "#fff",
          "dark": "#343a40",
          "light": "#dddddd",
          "main": "#6c757d",
        },
        "info": Object {
          "contrastText": "#fff",
          "dark": "#ba000d",
          "light": "#33c9dc",
          "main": "#00bcd4",
        },
        "isIphoneX": false,
        "platform": "ios",
        "primary": Object {
          "contrastText": "#fff",
          "dark": "#1769aa",
          "light": "#4dabf5",
          "main": "#2196f3",
        },
        "purple": Object {
          "contrastText": "#fff",
          "dark": "#6d1b7b",
          "light": "#af52bf",
          "main": "#9c27b0",
        },
        "secondary": Object {
          "contrastText": "#fff",
          "dark": "#b26a00",
          "light": "#ffac33",
          "main": "#ff9800",
        },
        "success": Object {
          "contrastText": "#fff",
          "dark": "#618833",
          "light": "#a2cf6e",
          "main": "#8bc34a",
        },
        "warning": Object {
          "contrastText": "#fff",
          "dark": "#b28704",
          "light": "#ffcd38",
          "main": "#ffc107",
        },
      }
    }
  >
    <Unknown>
      Title
    </Unknown>
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown>
        Title
      </Unknown>,
      "theme": Object {
        "danger": Object {
          "contrastText": "#fff",
          "dark": "#aa2e25",
          "light": "#f6685e",
          "main": "#f44336",
        },
        "deviceHeight": 750,
        "deviceWidth": 1334,
        "grey": Object {
          "contrastText": "#fff",
          "dark": "#343a40",
          "light": "#dddddd",
          "main": "#6c757d",
        },
        "info": Object {
          "contrastText": "#fff",
          "dark": "#ba000d",
          "light": "#33c9dc",
          "main": "#00bcd4",
        },
        "isIphoneX": false,
        "platform": "ios",
        "primary": Object {
          "contrastText": "#fff",
          "dark": "#1769aa",
          "light": "#4dabf5",
          "main": "#2196f3",
        },
        "purple": Object {
          "contrastText": "#fff",
          "dark": "#6d1b7b",
          "light": "#af52bf",
          "main": "#9c27b0",
        },
        "secondary": Object {
          "contrastText": "#fff",
          "dark": "#b26a00",
          "light": "#ffac33",
          "main": "#ff9800",
        },
        "success": Object {
          "contrastText": "#fff",
          "dark": "#618833",
          "light": "#a2cf6e",
          "main": "#8bc34a",
        },
        "warning": Object {
          "contrastText": "#fff",
          "dark": "#b28704",
          "light": "#ffcd38",
          "main": "#ffc107",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Title",
      },
      "ref": null,
      "rendered": "Title",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown>
          Title
        </Unknown>,
        "theme": Object {
          "danger": Object {
            "contrastText": "#fff",
            "dark": "#aa2e25",
            "light": "#f6685e",
            "main": "#f44336",
          },
          "deviceHeight": 750,
          "deviceWidth": 1334,
          "grey": Object {
            "contrastText": "#fff",
            "dark": "#343a40",
            "light": "#dddddd",
            "main": "#6c757d",
          },
          "info": Object {
            "contrastText": "#fff",
            "dark": "#ba000d",
            "light": "#33c9dc",
            "main": "#00bcd4",
          },
          "isIphoneX": false,
          "platform": "ios",
          "primary": Object {
            "contrastText": "#fff",
            "dark": "#1769aa",
            "light": "#4dabf5",
            "main": "#2196f3",
          },
          "purple": Object {
            "contrastText": "#fff",
            "dark": "#6d1b7b",
            "light": "#af52bf",
            "main": "#9c27b0",
          },
          "secondary": Object {
            "contrastText": "#fff",
            "dark": "#b26a00",
            "light": "#ffac33",
            "main": "#ff9800",
          },
          "success": Object {
            "contrastText": "#fff",
            "dark": "#618833",
            "light": "#a2cf6e",
            "main": "#8bc34a",
          },
          "warning": Object {
            "contrastText": "#fff",
            "dark": "#b28704",
            "light": "#ffcd38",
            "main": "#ffc107",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Title",
        },
        "ref": null,
        "rendered": "Title",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
